packages/flutter/lib/src/widgets/ticker_provider.dart

```diff
@@ -0,0 +1,210 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/scheduler.dart';
+
+import 'framework.dart';
+
+export 'package:flutter/scheduler.dart' show TickerProvider;
+
+/// Enables or disables tickers (and thus animation controllers) in the widget
+/// subtree.
+///
+/// This only works if [AnimationController] objects are created using
+/// widget-aware ticker providers. For example, using a
+/// [TickerProviderStateMixin] or a [SingleTickerProviderStateMixin].
+class TickerMode extends InheritedWidget {
+  TickerMode({ Key key, this.enabled, Widget child }) : super(key: key, child: child);
+
+  /// The current ticker mode of this subtree.
+  ///
+  /// If true, then tickers in this subtree will tick.
+  ///
+  /// If false, then tickers in this subtree will not tick. Animations driven by
+  /// such tickers are not paused, they just don't call their callbacks. Time
+  /// still elapses.
+  final bool enabled;
+
+  /// Whether tickers in the given subtree should be enabled or disabled.
+  ///
+  /// This is used automatically by [TickerProviderStateMixin] and
+  /// [SingleTickerProviderStateMixin] to enable and disable their tickers.
+  ///
+  /// In the absence of a [TickerMode] widget, this function defaults to true.
+  static bool of(BuildContext context) {
+    TickerMode widget = context.inheritFromWidgetOfExactType(TickerMode);
+    return widget?.enabled ?? true;
+  }
+
+  @override
+  bool updateShouldNotify(TickerMode old) => enabled != old.enabled;
+
+  @override
+  void debugFillDescription(List<String> description) {
+    super.debugFillDescription(description);
+    description.add('mode: ${ enabled ? "enabled" : "disabled" }');
+  }
+}
+
+/// Provides a single [Ticker] that is configured to only tick while the current
+/// tree is enabled, as defined by [TickerMode].
+///
+/// To create the [AnimationController] in a [State] that only uses a single
+/// [AnimationController], mix in this class, then pass `tickerProvider: this`
+/// to the animation controller constructor.
+///
+/// This mixin only supports vending a single ticker. If you might have multiple
```

Comment:
they are both fine. "vending" implies more of a transactional operation which is why i used that.